// nodemailer/sendmail.js
const path = require('path');
require('dotenv').config({path: path.resolve("./nodemailer/node_modules/dotenv", "config.env")});



const nodemailer = require('nodemailer');


function sendmail(mailList, token) {

   // console.log("Hey dirname ",__dirname);
   // console.log("Hey cwd ",process.cwd());
   // console.log("Hey mailList", mailList)

  //  let creds = token.split("/");
  //  console.log(creds);
  //  IDvoting = creds[0];
  //  IDvoter = creds[1];

   let transport = nodemailer.createTransport({
      host: 'smtp.gmail.com',
      port: 465,
      secure: true,
      auth: {
         user: process.env.EMAIL_USERNAME,
         pass: process.env.EMAIL_PASSWORD

      }
   });

   // var mailList = [
   //    'zgsakellariou@gmail.com',
   // ];

   let url = "http://localhost:3000/";
   let urlWithToken = url + "?" + "token=" + token;

   let mailOptions = {
      from: 'iotfederations@gmail.com', // Sender address
      to: mailList, // List of recipients
      subject: 'IoTFeds Values Your Opinion', // Subject line
      // html: `<p>Dear IoTFeds User,</p><p>Please take a moment to cast your vote through this <a href="http://localhost:3000/">link</a>!</p><p>Your voting token is ${token}`,
      html: `<p>Dear IoTFeds User,</p><br><p>Please take a moment to cast your vote through this link: <a href=${urlWithToken}>${urlWithToken}</a>`,
      attachments: [
         { path: path.resolve("./nodemailer/node_modules/nodemailer/img", "iotfeds-logo.png") }
      ]
   };

   transport.sendMail(mailOptions, function(err, info) {
      if (err) {
        console.log(err)
      } else {
        console.log(info);
      }
   });
}

module.exports = {sendmail};
